eng/illink.targets

```diff
@@ -30,7 +30,8 @@
          we might be able to use built-in functionality instead of a packagereference.
          -->
     <_RequiresILLinkPack>false</_RequiresILLinkPack>
-    <ILLinkTasksAssembly>$(ToolsILLinkDir)$(NetCoreAppToolCurrent)/ILLink.Tasks.dll</ILLinkTasksAssembly>
+    <ILLinkTasksAssembly Condition="'$(MSBuildRuntimeType)' == 'Core'">$(ToolsILLinkDir)$(NetCoreAppToolCurrent)\ILLink.Tasks.dll</ILLinkTasksAssembly>
+    <ILLinkTasksAssembly Condition="'$(MSBuildRuntimeType)' != 'Core'">$(ToolsILLinkDir)$(NetFrameworkToolCurrent)\ILLink.Tasks.dll</ILLinkTasksAssembly>
```

Comment:
The VS build node `msbuild.exe` processes are long-lived so it could break if you ever modified `ILLink.Tasks` _between_ builds too. `TaskHostFactory` is what we generally recommend for any task that's built and used in the same repo.